tests: lpuart: uart_async_api: build failure for mimxrt10xx series

**Describe the bug**
Build failure

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 -  mimxrt10xx, mr_canhubk3, mimxrt11xx series
 - What have you tried to diagnose or workaround this issue?
 - NA
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 
```
9320907fdf1f6dc436c426d4d31f0a0f6d2d2e8f is the first bad commit
commit 9320907fdf1f6dc436c426d4d31f0a0f6d2d2e8f
Author: Declan Snyder <declan.snyder@nxp.com>
Date:   Wed Apr 24 19:37:48 2024 -0500

    drivers: uart_mcux_lpuart: Fix APIs used together
    
    Handle controller instances wanting to use different
    API instances, need to change init flow so that they dont
    interfere with each other, trick for now is returning
    either the interrupt driven or async apis as not enabled
    if the other has already been used on that controller.
    
    Previously, there was an issue where enabling CONFIG_UART_ASYNC_API
    would overwrite the init even if controller was meant to use
    interrupt driven (as kconfig is global to affect all the controllers)
    
    Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

 drivers/serial/uart_mcux_lpuart.c | 193 ++++++++++++++++++++++++++++----------
 1 file changed, 143 insertions(+), 50 deletions(-)
```

**To Reproduce**
Steps to reproduce the behavior:

scripts/twister -p mimxrt10160_evk -T mimxrt1170_evk@B/mimxrt1176/cm7 --build-only

**Expected behavior**
Build pass

**Impact**
build CI

**Logs and console output**

```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/serial/uart_mcux_lpuart.c:1030:12: error: 'mcux_lpuart_configure_async' defined but not used [-Werror=unused-function]
 1030 | static int mcux_lpuart_configure_async(const struct device *dev)
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v3.6.0-3395-g1ae357850d5c

--- a/drivers/serial/uart_mcux_lpuart.c
+++ b/drivers/serial/uart_mcux_lpuart.c
@@ -714,10 +714,9 @@ static int mcux_lpuart_callback_set(const struct device *dev, uart_callback_t ca
 	data->callback = NULL;
 	data->cb_data = NULL;
 	data->api_type = LPUART_ASYNC;
-	return mcux_lpuart_configure_async(dev);
-#else
-	return 0;
 #endif
+
+	return mcux_lpuart_configure_async(dev);
 }
 
 static int mcux_lpuart_tx(const struct device *dev, const uint8_t *buf, size_t len,

